cmd/minishift/cmd/root.go

```diff
@@ -174,18 +214,21 @@ func init() {
 	hasEnabledExperimental = enableExperimental
 
 	RootCmd.PersistentFlags().Bool(showLibmachineLogs, false, "Show logs from libmachine.")
+	RootCmd.PersistentFlags().String(profile, "", "Profile name")
 	RootCmd.AddCommand(configCmd.ConfigCmd)
 	RootCmd.AddCommand(cmdOpenshift.OpenShiftCmd)
 	RootCmd.AddCommand(hostfolderCmd.HostfolderCmd)
 	RootCmd.AddCommand(addon.AddonsCmd)
 	RootCmd.AddCommand(image.ImageCmd)
+	RootCmd.AddCommand(cmdProfile.ProfileCmd)
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
 	logDir := pflag.Lookup("log_dir")
 	if !logDir.Changed {
 		logDir.Value.Set(constants.MakeMiniPath("logs"))
 	}
 	viper.BindPFlags(RootCmd.PersistentFlags())
 	cobra.OnInitialize(initConfig)
```

Comment:
@LalatenduMohanty have you tested the use of the persistent config? For all I can tell from looking at this code, the code initialises against the default location instead of taking the profile into account. _init_ is called after _vars_ are initialised and before main. This means this method is called long before anything else happens. It then registered a callback for cobra.OnInitialize which is most likely also called before _PersistentPreRun_. So when you finally get around changing the various directories Viper is already initialised with the wrong/default _config.json_. Correct me from wrong.

root.go needs to change in various forms.